File: app/src/main/java/faith/changliu/androiddemo/data/GameDataProvider.java
package faith.changliu.androiddemo.data;

import android.content.Context;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Iterator;

import faith.changliu.androiddemo.data.entities.Game;
import faith.changliu.androiddemo.helpers.CommonUtils;

public class GameDataProvider {

	/**
	 * Get games from fake data from fakeData.json
	 */
	public static ArrayList<Game> getGamesFromFakeData(Context ctx) {
		String jsonString = CommonUtils.readStringFromAssets(ctx, "fakeData.json");
		return convertStringToGames(jsonString);
	}

	/**
	 * Try convert JSON string to games list
	 *
	 * @param jsonString String input
	 *
	 * @return Returns a new ArrayList<Game>
	 */
	private static ArrayList<Game> convertStringToGames(String jsonString) {
		ArrayList<Game> result = new ArrayList<>();
		JSONObject jsonObject;
		try {
			jsonObject = new JSONObject(jsonString);
			Iterator<String> keys = jsonObject.keys();
			while (keys.hasNext()) {
				JSONObject newGameJSON = jsonObject.getJSONObject(keys.next());
				result.add(new Game(newGameJSON));
			}
		} catch (JSONException e) {
			e.printStackTrace();
		}

		return result;
	}
}

File: app/src/main/java/faith/changliu/androiddemo/data/GameDataProvider.kt
package faith.changliu.androiddemo.data

import faith.changliu.androiddemo.AppContext
import faith.changliu.androiddemo.helpers.CommonUtils
import faith.changliu.androiddemo.helpers.FILE_FAKE_DATA
import org.json.JSONException
import org.json.JSONObject

object GameDataProvider {

	/**
	 * Get games from fake data
	 *
	 * @param fileName Name of the file to be read inside assets, default to fakeData.json
	 */
	fun getGamesFromFakeData(fileName: String = FILE_FAKE_DATA): ArrayList<Game> {
		val jsonString = CommonUtils.readStringFromAssets(AppContext.instance, fileName)
		return convertStringToGames(jsonString)
	}

	/**
	 * Try convert JSON string to games list
	 *
	 * @param jsonString String input
	 *
	 * @return Returns a new ArrayList<Game>
	 */
	private fun convertStringToGames(jsonString: String): ArrayList<Game> {
		val result = arrayListOf<Game>()
		try {
			val jsonObject = JSONObject(jsonString)
			jsonObject.keys().forEach { key ->
				(jsonObject[key] as? JSONObject)?.let {
					result.add(Game(it))
				}
			}
		} catch (ex: JSONException) {
			ex.printStackTrace()
		} catch (ex: Exception) {
			ex.printStackTrace()
		}

		return result
	}
}